pkg/security/authenticators/basic_authentication.go

```diff
@@ -57,7 +70,77 @@ func (a *Basic) Authenticate(request *web.Request) (*web.UserContext, httpsec.De
 		return nil, httpsec.Deny, fmt.Errorf("provided credentials are invalid")
 	}
 
-	bytes, err := json.Marshal(platform)
+	return buildResponse(username, platform)
+}
+
+//BasicOSBAuthenticator attempts to authenticate basic auth requests with provided broker platform credentials
+func BasicOSBAuthenticator(request *web.Request, repository storage.Repository, username, password string) (*web.UserContext, httpsec.Decision, error) {
+	ctx := request.Context()
+	log.C(ctx).Debugf("Attempting to authenticate broker platform credentials")
+
+	brokerID, ok := request.PathParams[osb.BrokerIDPathParam]
+	if !ok {
+		return nil, httpsec.Abstain, fmt.Errorf("could not get authenticate OSB request: brokerID path parameter not found")
+	}
+
+	byBrokerID := query.ByField(query.EqualsOperator, "broker_id", brokerID)
+	byUsername := query.ByField(query.EqualsOperator, "username", username)
+
+	credentialsList, err := repository.List(ctx, types.BrokerPlatformCredentialType, byBrokerID, byUsername)
+	if err != nil {
+		return nil, httpsec.Abstain, fmt.Errorf("could not get credentials entity from storage: %s", err)
+	}
+
+	useOldCredentials := false
```

Comment:
probably instead of this bool you can have the actual struct for holding credentials here and just use it with the correct values assigned later on. Then you can remove `if useOldCredentials {
		passwordHash = credentials.OldPasswordHash
	}`